src/iceberg/expected.h

```diff
@@ -122,7 +122,7 @@ inline constexpr bool is_error_type_valid_v = is_error_type_valid<T>::value;
 }  // namespace expected_detail
 
 template <class E>
-class ICEBERG_EXPORT [[nodiscard]] unexpected {
```

Comment:
I have to remove these to fix https://github.com/apache/iceberg-cpp/actions/runs/15342734208/job/43172206905. I think adding `ICEBERG_EXPORT` here makes all `Result<E>` defined in the `libiceberg-bundle` as  import symbols and the linker cannot find them anywhere. @zhjwpku @lidavidm

Reply:
Hmm, if it works it works. I really don't get how the Windows model works.